### PR TITLE
1434 [Transactions] Environment definition execution

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the definition yaml execution support for environment files, same logic as for tests, if the id exits we update, if not we create a new brand entry to the DB.

## Changes

- Adds execution support for environment definition files

## Fixes

- #1434 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/78716627311d45209f14214d6cb0c961 [NO SOUND]

### CODE REVIEW

**Review by jorgeepc — APPROVED** [review]
🔥 LGTM!

File: server/model/yaml/environment.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+			Key:   ev.Key,
+			Value: ev.Value,
+		})
+	}
+
+	return mevs
+}
+
+func (e Environment) Validate() error {
+	if e.Name == "" {
```
Comment (mathnogueira) [inline server/model/yaml/environment.go:36]: Shouldn't we validate if the variables names are set?

File: server/http/controller.go [hunk 2]
```diff
@@ -484,9 +484,38 @@ func (c *controller) ExecuteDefinition(ctx context.Context, testDefinition opena
 		return c.executeTest(ctx, test.Model(), testDefinition.RunInformation)
 	}
 
+	if environment, err := def.Environment(); err == nil {
+		return c.createEnvFromDefinition(ctx, environment)
```
Comment (mathnogueira) [inline server/http/controller.go:488]: This is returning the wrong response. It should return the testId, not te env ID.

**Review by mathnogueira — CHANGES_REQUESTED** [review]

File: server/http/controller.go (hunk 2 above)
Comment (xoscar) [reply server/http/controller.go:488]: I'm a little confused here, when creating an environment from a definition file there won't be any info regarding the test so my idea was to return the environment id as the creation output, if that's not expected, do you think I should add a new field to the response to be environment id instead?

File: server/model/yaml/environment.go (hunk 1 above)
Comment (xoscar) [reply server/model/yaml/environment.go:36]: Done 😄

File: server/http/controller.go (hunk 2 above)
Comment (mathnogueira) [reply server/http/controller.go:488]: Maybe this shouldn't be here then. Maybe the CLI should create the env before running the test and we only reference the environment here. Another thing, even if we leave this here, shouldn't this happen before the `executeTest` function call?

To execute a test we still need the transaction to exist.

Comment (mathnogueira) [reply server/http/controller.go:488]: Oh I see what you did there. It is only be executed if the resource is an environment. Ignore my comments. Makes sense.

**Review by mathnogueira — APPROVED** [review]
Looks good

**Pull request merged**